sphinx/builders/__init__.py

```diff
@@ -473,6 +483,9 @@ def read(self):
             if retval is not None:
                 docnames.extend(retval)
 
+        # workaround: marked as okay to call builder.read() twice in same process
+        self.env.config_status = CONFIG_OK
```

Comment:
I feel touching status of env object directly here is not good. But I also feel ugly to add a new method...

Reply:
This seems to somehow influence the bug reported in #12120 - I'm not sure exactly how/why yet, but it means that a test that should refresh its environment settings does not, and reads some stale/inaccurate env config.